docker-app/qfieldcloud/core/tests/test_secret_packages.py

```diff
@@ -0,0 +1,245 @@
+                role=OrganizationMember.Roles.ADMIN,
+            ),
+        ]
+
+        self.o1.members.bulk_create(members)
+
+        collaborators = [
+            ProjectCollaborator(
+                project=self.p1,
+                collaborator=self.u2,
+                role=ProjectCollaborator.Roles.ADMIN,
+            ),
+        ]
+
+        self.p1.direct_collaborators.bulk_create(collaborators)
+
+    def _create_secret(self, **kwargs) -> Secret:
+        return Secret.objects.create(
+            type=Secret.Type.ENVVAR,
+            **kwargs,
+        )
+
+    def test_create_user_level_packages(self):
+        # upload data & QGIS project files to the project.
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "bumblebees.gpkg",
+            io.FileIO(testdata_path("bumblebees.gpkg"), "rb"),
+        )
+
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "simple_bumblebees.qgs",
+            io.FileIO(testdata_path("simple_bumblebees.qgs"), "rb"),
+        )
+
+        # upload some attachments files to the project.
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "DCIM/1.jpg",
+            io.FileIO(testdata_path("DCIM/1.jpg"), "rb"),
+        )
+
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "DCIM/2.jpg",
+            io.FileIO(testdata_path("DCIM/2.jpg"), "rb"),
+        )
+
+        # create secrets for project and assigned to users
+        self._create_secret(name="SECRET", project=self.p1, value="p1")
+        self._create_secret(
+            name="SECRET", project=self.p1, assigned_to=self.u1, value="u1"
+        )
+        self._create_secret(
+            name="SECRET", project=self.p1, assigned_to=self.u2, value="u2"
+        )
+
+        self.p1.refresh_from_db()
+
+        # create a package for the two users
+        repackage(self.p1, self.u1)
+        repackage(self.p1, self.u2)
+
+        wait_for_project_ok_status(self.p1)
+        self.p1.refresh_from_db()
+
+        latest_package_jobs_qs = self.p1.latest_package_jobs()
+
+        self.assertEquals(latest_package_jobs_qs.count(), 2)
+
+        latest_packages_users = latest_package_jobs_qs.values_list(
+            "created_by", flat=True
+        )
+
+        self.assertIn(self.u1.id, latest_packages_users)
+        self.assertIn(self.u2.id, latest_packages_users)
+
+    def test_create_org_level_packages(self):
+        # upload data & QGIS project files to the project.
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "bumblebees.gpkg",
+            io.FileIO(testdata_path("bumblebees.gpkg"), "rb"),
+        )
+
+        self._upload_file(
+            self.u1,
+            self.p1,
+            "simple_bumblebees.qgs",
+            io.FileIO(testdata_path("simple_bumblebees.qgs"), "rb"),
+        )
+
+        # upload some attachments files to the project.
```

Comment:
Just wondering do we need attachments at all? If yes, shall we assert something with them? If not, I believe we can remove them.